StartingExample/MapScreen.cs

```diff
@@ -0,0 +1,85 @@
+﻿using GoRogue;
+using GoRogue.GameFramework;
+using GoRogue.MapGeneration;
+using GoRogue.MapViews;
+using Microsoft.Xna.Framework;
+using SadConsole;
+using XnaRect = Microsoft.Xna.Framework.Rectangle;
+
+namespace StartingExample
+{
+    class MapScreen : ContainerConsole
+    {
+        public ExampleMap Map { get; }
+        public ScrollingConsole MapRenderer { get; }
+        //Generate a map and display it.  Could just as easily pass it into
+        public MapScreen(int mapWidth, int mapHeight, int viewportWidth, int viewportHeight)
+        {
+            Map = GenerateDungeon(mapWidth, mapHeight);
+
+            // Get a console that's set up to render the map, and add it as a child of this container so it renders
+            MapRenderer = Map.CreateRenderer(new XnaRect(0, 0, viewportWidth, viewportHeight), SadConsole.Global.FontDefault);
+            Children.Add(MapRenderer);
+            Map.ControlledGameObject.IsFocused = true; // Set player to receive input, since in this example the player handles movement
+
+            // Set up to recalculate FOV and set camera position appropriately when the player moves.  Also make sure we hook the new
+            // Player if that object is reassigned.
+            Map.ControlledGameObject.Moved += Player_Moved;
+            Map.ControlledGameObjectChanged += ControlledGameObjectChanged;
+
+            // Calculate initial FOV and center camera
+            Map.CalculateFOV(Map.ControlledGameObject.Position, Map.ControlledGameObject.FOVRadius, Radius.SQUARE);
+            MapRenderer.CenterViewPortOnPoint(Map.ControlledGameObject.Position);
+        }
+
+        private void ControlledGameObjectChanged(object s, ControlledGameObjectChangedArgs e)
+        {
+            if (e.OldObject != null)
+                e.OldObject.Moved -= Player_Moved;
+
+            ((BasicEntity)s).Moved += Player_Moved;
```

Comment:
I think this is supposed to be the map that is the sender.

```suggestion
            ((BasicMap)s)ControlledGameObject.Moved += Player_Moved;
```